### PR TITLE
Add Promotion Banner

### PR DESCRIPTION
**Pull request by thayannevls**
#### What is the purpose of this pull request?
Create promotion banner component to promete web app install and push notifications.

#### What problem is this solving?
We need to promote our pwa features.

#### How should this be manually tested?
[Workspace](https://thay--storecomponents.myvtex.com/checkout/orderPlaced/?og=949500478939)

#### Types of changes

* [ ] Bug fix (a non-breaking change which fixes an issue)
* [x] New feature (a non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/PromotionBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  PUSH_NOTIFICATION_TITLE: 'store/promotion-banner.push-notification.title',
+  PUSH_NOTIFICATION_DESCRIPTION: 'store/promotion-banner.push-notification.description',
+  PUSH_NOTIFICATIONS_ACCEPT_BUTTON_LABEL: 'store/promotion-banner.push-notification.button-accept.label',
+  DISMISS_BUTTON_LABEL: 'store/promotion-banner.button-dismiss.label'
+}
+
+type Props = PropTypes.InferProps<typeof propTypes>
+
+const PromotionBanner: FC<Props & InjectedIntlProps> = ({ intl, type = 'install' }) => {
+  const { showInstallPrompt = null } = usePWA() || {}
```
Comment (luciannojunior) [inline react/PromotionBanner.tsx]: Could it really be null or is it just a TS inconvenience? If so, add a check on [#39](https://github.com/vtex-apps/pwa-components/pull/3/files#diff-5c3386809b77f6d98312956eb41f6355R39)

File: react/PromotionBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+  const handleDismiss = async () => {
```
Comment (luciannojunior) [inline react/PromotionBanner.tsx]: Is the `async` necessary?

Comment (luciannojunior) [reply react/PromotionBanner.tsx]: And also `useCallback`

File: react/PromotionBanner.tsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    if(type === 'install')
+      setWebAppData('appInstallDismissed')
+    else
+      setWebAppData('pushNotificationsDismissed')
+  }
+
+  const handleAccept = () => {
```
Comment (luciannojunior) [inline react/PromotionBanner.tsx]: Should you `useCallback` here?

File: react/PromotionBanner.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    if(type === 'install')
```
Comment (luciannojunior) [inline react/PromotionBanner.tsx]: Didn't the lint/prettier complain about not using curly-braces?

File: react/PromotionBanner.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+type Props = PropTypes.InferProps<typeof propTypes>
+
+const PromotionBanner: FC<Props & InjectedIntlProps> = ({ intl, type = 'install' }) => {
+  const { showInstallPrompt = null } = usePWA() || {}
+
+  const handleDismiss = async () => {
+    if(type === 'install')
+      setWebAppData('appInstallDismissed')
```
Comment (luciannojunior) [inline react/PromotionBanner.tsx]: It might makes sense to put all these string in the `CONSTANTS`

File: react/PromotionBanner.tsx (hunk 1 above)
Comment (thayannevls) [reply react/PromotionBanner.tsx]: It's just a TS inconvenience 😅

File: react/PromotionBanner.tsx (hunk 2 above)
Comment (thayannevls) [reply react/PromotionBanner.tsx]: `setWebAppData` returns a promise, that's I'm using `async`. Just changed to use `useCallback`, thank you for the tip!

File: react/PromotionBanner.tsx (hunk 5 above)
Comment (thayannevls) [reply react/PromotionBanner.tsx]: Done!

File: docs/PromotionBanner.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+:loudspeaker: **Disclaimer:** Don't fork this project, use, contribute, or open issue with your feature request.
+
+## Table of Contents
+
+- [Usage](#usage)
+  - [Blocks API](#blocks-api)
+    - [Configuration](#configuration)
+
+## Usage
```
Comment (lucasecdb) [inline docs/PromotionBanner.md]: I think you can remove these lines

**Pull request merged**